test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs

```diff
@@ -184,6 +187,51 @@ public Task RemoteIPv6Address()
             }
         }
 
+        [Fact]
+        public async Task ConnectionResetAbortsRequest()
+        {
+            const int connectionErrorEventId = 14; // from KestrelTrace.cs
+
+            var testSink = new TestSink(write => write.EventId.Id == connectionErrorEventId);
+            var requestAbortedAcquired = new CancellationTokenSource();
+            var builder = new WebHostBuilder()
+                .UseLoggerFactory(new TestLoggerFactory(testSink, true))
+                .UseKestrel()
+                .UseUrls($"http://127.0.0.1:0")
+                .Configure(app => app.Run(context =>
+                {
+                    return Task.FromResult(0);
```

Comment:
Shouldn't that be the case for any connection error? Why aren't we doing that already?

If that's further work not specific to ECONNRESET I'll create a separate task to address this.